### PR TITLE
RF: fix NCCL bug

### PR DESCRIPTION
**Pull request by wbo4958**
I tried running RandomForestClassifier on the spark standalone cluster with 2 workers, each of which has 1 GPU. But I got the below NCCL error.

``` console
_py4j.protocol.Py4JJavaError: An error occurred while calling z:org.apache.spark.api.python.PythonRDD.collectAndServe.
: org.apache.spark.SparkException: Job aborted due to stage failure: Could not recover from a failed barrier ResultStage. Most recent failure reason: Stage failed because barrier task ResultTask(1, 1) finished unsuccessfully.
org.apache.spark.api.python.PythonException: Traceback (most recent call last):
File "/tmp/spark-93495d91-6f13-4b97-b04e-36c9b713cc7a/userFiles-c6003233-1a32-4d54-b627-8e913b171274/sparkcuml.zip/spark_rapids_ml/core.py", line 419, in _train_udf
File "./sparkcuml.zip/spark_rapids_ml/common/cuml_context.py", line 104, in _enter_
self._nccl_comm.init(self._nranks, self._nccl_unique_id, self._rank)
File "nccl.pyx", line 151, in raft_dask.common.nccl.nccl.init
RuntimeError: NCCL_ERROR: b'remote process exited or there was a network error'_
```

It turned out #143 has changed the API.

### CODE REVIEW

File: src/spark_rapids_ml/tree.py [hunk 1]
```diff
@@ -272,8 +272,8 @@ def _out_schema(self) -> Union[StructType, str]:
             fields.append(StructField("num_classes", IntegerType(), False))
         return StructType(fields)
 
-    def _enable_nccl(self) -> bool:
-        return False
+    def _require_nccl_ucx(self) -> Tuple[bool, bool]:
+        return False, False
```
[the remark this answers is not in the document]
Comment (leewyang) [reply src/spark_rapids_ml/tree.py:276]: Also, should this be two separate methods/flags, i.e. `_require_nccl` and `_require_ucx`, or are they only meaningful as a pair/tuple?

[the remark this answers is not in the document]
Comment (eordentlich) [reply src/spark_rapids_ml/tree.py:276]: They were originally separate.  @wbo4958 suggested to merge them but the rf code wasn't updated to reflect the change.

**Review by eordentlich — APPROVED** [review]
👍

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**